kernel/Kconfig.smp

```diff
@@ -73,6 +73,21 @@ config TRACE_SCHED_IPI
 	depends on SCHED_IPI_SUPPORTED
 	depends on MP_MAX_NUM_CPUS>1
 
+config IPI_OPTIMIZE
+	bool "Optimize IPI delivery"
+	default y
```

Comment:
Pretty sure you want n here, no?  Most RTOS apps don't do a lot of scheduling and want the smaller code size and O(1) scheduler performance.

Reply:
Oops. This is a remnant from an earlier draft. I thought I had changed it 'n', but apparently not. Thanks for the catch!